src/plugins/dashboard/public/dashboard_app/dashboard_app.tsx

```diff
@@ -119,9 +119,9 @@ export function DashboardApp({
    */
   const getCreationOptions = useCallback((): Promise<DashboardCreationOptions> => {
     const searchSessionIdFromURL = getSearchSessionIdFromURL(history);
+    const initialUrlState = loadAndRemoveDashboardState(kbnUrlStateStorage);
     const getInitialInput = () => {
       const stateFromLocator = loadDashboardHistoryLocationState(getScopedHistory);
```

Comment:
Why was this moved?

Reply:
The function `loadAndRemoveDashboardState` can only be executed once. After executed, the search query will be removed, and the result will be empty.

So I move `loadAndRemoveDashboardState` out of `getInitialInput` (which can be executed many times) to keep the result is correct.